projects/js-packages/ai-client/src/hooks/use-ai-suggestions/index.ts

```diff
@@ -0,0 +1,365 @@
+	/*
+	 * The suggestion.
+	 */
+	suggestion: string;
+
+	/*
+	 * The error.
+	 */
+	error: SuggestionErrorProps | undefined;
+
+	/*
+	 * The post ID.
+	 */
+	postId: number;
+
+	/*
+	 * Whether the request is in progress.
+	 */
+	requestingState: RequestingStateProp;
+
+	/*
+	 * The event source.
+	 */
+	eventSource: SuggestionsEventSource | undefined;
+
+	/*
+	 * The request handler.
+	 */
+	request: ( prompt: Array< PromptItemProps > ) => Promise< void >;
+};
+
+const debug = debugFactory( 'jetpack-ai-client:use-suggestion' );
+
+/**
+ * Get the error data for a given error code.
+ *
+ * @param {SuggestionErrorCode} errorCode - The error code.
+ * @returns {SuggestionErrorProps}          The error data.
+ */
+function getErrorData( errorCode: SuggestionErrorCode ): SuggestionErrorProps {
+	switch ( errorCode ) {
+		case ERROR_QUOTA_EXCEEDED:
+			return {
+				code: ERROR_QUOTA_EXCEEDED,
+				message: __( 'You have reached the limit of requests for this site.', 'jetpack-ai-client' ),
+				severity: 'info',
+			};
+		case ERROR_UNCLEAR_PROMPT:
+			return {
+				code: ERROR_UNCLEAR_PROMPT,
+				message: __( 'Your request was unclear. Mind trying again?', 'jetpack-ai-client' ),
+				severity: 'info',
+			};
+		case ERROR_SERVICE_UNAVAILABLE:
+			return {
+				code: ERROR_SERVICE_UNAVAILABLE,
+				message: __(
+					'Jetpack AI services are currently unavailable. Sorry for the inconvenience.',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		case ERROR_MODERATION:
+			return {
+				code: ERROR_MODERATION,
+				message: __(
+					'This request has been flagged by our moderation system. Please try to rephrase it and try again.',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		case ERROR_NETWORK:
+			return {
+				code: ERROR_NETWORK,
+				message: __(
+					'It was not possible to process your request. Mind trying again?',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		default:
+			return {
+				code: ERROR_NETWORK,
+				message: __(
+					'It was not possible to process your request. Mind trying again?',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+	}
+}
+
+/**
+ * React custom hook to get suggestions from AI,
+ * by hitting the query endpoint.
+ *
+ * @param {useAiSuggestionsOptions} options - The options for the hook.
+ * @returns {useAiSuggestionsProps}           The props for the hook.
+ */
+export default function useAiSuggestions( {
+	prompt,
+	autoRequest = false,
+	askQuestionOptions = {},
+	onSuggestion,
+	onDone,
+	onError,
+}: useAiSuggestionsOptions = {} ): useAiSuggestionsProps {
+	const [ requestingState, setRequestingState ] = useState< RequestingStateProp >( 'init' );
+	const [ suggestion, setSuggestion ] = useState< string >( '' );
+	const [ error, setError ] = useState< SuggestionErrorProps >();
+
+	// Try to pick the post ID to populate the askQuestion request.
+	const postId = useSelect( select => select( 'core/editor' ).getCurrentPostId(), [] );
```

Comment:
I am not sure we should tie the hook to the editor context. As far as I read the code, this is the only connection to the editor, so if we want to use the hook on a different place and yet set the postId somehow (like the attachment ID for a VideoPress video, for example), it would not be possible.

Would be possible to have this info set as a parameter?

Reply:
https://github.com/Automattic/jetpack/pull/32022/commits/e19d7b4f2a855ea9891743ad443e366b3bbc8782 and https://github.com/Automattic/jetpack/pull/32022/commits/6edcdb425f50b93b400603907034a4935574bbfc